File: HolyChildhood/ClientApp/src/app/shared/models/calendar.ts
export interface Calendar {
    id: number;
    name: string;
    events: Event[];
}

export interface Event {
    id: number;
    calendarId: number;
    title: string;
    description: string;
    location: string;
    notes: string;
    start: Date;
    end: Date;
    allDay: boolean;
    eventTypeId: number;
    eventTypeName: string;
    color: string;

    // Recurrence Data
    isRecurring: boolean;
    recurrenceId: string;
    recurrenceType: string;
    recurrenceMonthlyType: string;
    recurrenceMonthlyWeek: number;
    updateRecurrence: boolean;

    // support EventViewModel
    startYear: number;
    startMonth: number;
    startDay: number;
    startHour: number;
    startMinute: number;
    hasEndTime: boolean;
    endHour: number;
    endMinute: number;

    // other support properties
    isEditing: boolean;
    startTime: Date;
    endTime: Date;
}

File: HolyChildhood/Models/Calendar.cs
using System;
using System.Collections.Generic;

namespace HolyChildhood.Models
{
    public class Calendar
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public List<Event> Events { get; set; }
    }

    public class Event
    {
        public int Id { get; set; }
        public Calendar Calendar { get; set; }
        public string Title { get; set; }
        public DateTime BeginDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string Description { get; set; }
        public string Location { get; set; }
        public string Notes { get; set; }
        public bool AllDay { get; set; }
        public bool IsRecurring { get; set; }
        public int? EventTypeId { get; set; }
        public EventType EventType { get; set; }
        public Guid? RecurrenceId { get; set; }
    }

    public class EventType
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Color { get; set; }
    }

}